grove/amplification/amplification.py

```diff
@@ -40,27 +52,39 @@ def amplification_circuit(algorithm, oracle, qubits, num_iter):
         subspace.
     :param Sequence qubits: the qubits to operate on
     :param int num_iter: number of iterations of amplifications to run
+    :param bool decompose_diffusion: If True, decompose the Grover diffusion gate into two qubit
+     gates. If False, use a defgate to define the gate.
     :return: The amplified algorithm.
     :rtype: Program
     """
-    if num_iter <= 0:
-        raise ValueError("num_iter must be greater than zero")
-
     prog = pq.Program()
 
     uniform_superimposer = pq.Program().inst([H(qubit) for qubit in qubits])
     prog += uniform_superimposer
-
+    if decompose_diffusion:
+        diffusion_program = decomposed_diffusion_program(qubits)
+    else:
+        diffusion_program = diffusion_circuit(qubits)
```

Comment:
`decomposed_diffusion_program` vs `diffusion_circuit`, can we make the naming semantically similar so the intention is clear?

Reply:
Yep!